### PR TITLE
ltq-ptm: Update VR9 PTM firmware

### PR DESCRIPTION
**Pull request by pesintta**
After looking at various vendor GPL source code dumps I discovered that some of them contain updated versions of ltq-ptm driver which in turn contains an updated firmware code blob for PTM PPE which is extracted in this pull-request along with minimum set of version identifying bits from the updated driver.

I'm hopeful the firmware update would help with bugs like FS#494 where ptm tx engine somehow hangs after some varying time of operation.

Someone more in the know (@blogic or @hauke perhaps) might want to take a look to see if right fw was used and everything is done as it should.

### CODE REVIEW

File: package/kernel/lantiq/ltq-ptm/src/ifxmips_ptm_fw_regs_vdsl.h [hunk 1]
```diff
@@ -79,7 +79,18 @@ struct wan_tx_mib_table {
 
 #if defined(__BIG_ENDIAN)
 
-  struct fw_ver_id {
+  struct fw_ver_id_new {    //  @2000
+	/*  0 - 3h  */
+	unsigned int family         :4;
+    unsigned int package        :4;
+    unsigned int major          :8;
+    unsigned int middle         :8;
```
Comment (adschm) [inline package/kernel/lantiq/ltq-ptm/src/ifxmips_ptm_fw_regs_vdsl.h:87]: Any reason for this strange indent?

Comment (pesintta) [reply package/kernel/lantiq/ltq-ptm/src/ifxmips_ptm_fw_regs_vdsl.h:87]: I tried to religiously follow the coding conventions (weird indents, whitespace etc.) found in the original driver code used as a reference here which is... not very well formatted indeed.

Comment (dangowrt) [reply package/kernel/lantiq/ltq-ptm/src/ifxmips_ptm_fw_regs_vdsl.h:87]: @pesintta Please fix the indent, do `git commit -a --amend` and force-push onto the PR branch, so we can move forward with this. Thank you!

Comment (pesintta) [reply package/kernel/lantiq/ltq-ptm/src/ifxmips_ptm_fw_regs_vdsl.h:87]: @dangowrt thanks, I've now fixed the indent.

**Comment by dhewg** [comment]
I didn't have any issue with the old version, but this works fine too, so no regressions afaict ;)
`Tested-by: <Andre Heider a.heider@gmail.com>`

Tested on TP-LINK TD-W8970 using:
```
config vdsl 'dsl'
    option annex 'b'
    option tone 'bv'
    option xfer_mode 'ptm'
    option line_mode 'vdsl'
```

Before:
```
PTM 1.0.27    PTM (E1) firmware version 0.30
ifxmips_ptm: PTM init succeed
```
After:
```
Lantiq PTM driver, version 1.0.27, (c) 2001-2013 Lantiq Deutschland GmbH
    PTM (E1) firmware version 2.16.0
ifxmips_ptm: PTM init succeed
```

**Comment by adschm** [comment]
From reading the commit message, I'm not really sure whether we are allowed to add this. Where exactly did you get the code from? Did "Lantiq Deutschland GmbH" really release the code with a license that allows us to include it here?

**Comment by pesintta** [comment]
> From reading the commit message, I'm not really sure whether we are allowed to add this. Where exactly did you get the code from? Did "Lantiq Deutschland GmbH" really release the code with a license that allows us to include it here?

All the code can be found (among other releases) from TP-Link's Archer AX v1 GPL code release (https://static.tp-link.com/resources/gpl/ax50v1_GPL.tar.gz).

The updated ptm-driver is in its entirety in: 
./intel_GRX350/sdk/UGW-7.5.1.40/ugw/target/linux/lantiq/linux-3.10.104-sources/drivers/net/lantiq_ptm/

Every file used as a reference (including the firmware) has GPLv2-or-later boilerplate text and they all belong to a GPL-source code release done by a known vendor so I figured we would be allowed to use them.

I can of course refactor the dmesg print to resemble something that we used to have if that's better.

**Review by dhewg — APPROVED** [review]

**Comment by dhewg** [comment]
This can be closed, commited as 38646938a458d0d9eda7ea3d14e182892470efe8

**Pull request closed without merging**